clang/include/clang/Basic/SyclOptReportHandler.h

```diff
@@ -0,0 +1,57 @@
+//===---------------------- SyclOptReport.h ---------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Defines clang::SyclOptReport class.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_BASIC_SYCLOPTREPORTHANDLER_H
+#define LLVM_CLANG_BASIC_SYCLOPTREPORTHANDLER_H
+
+#include "clang/Basic/SourceLocation.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/StringRef.h"
+
+namespace clang {
+
+class FunctionDecl;
+
+class SyclOptReportHandler {
+private:
+  struct OptReportInfo {
+    StringRef KernelArgName;
```

Comment:
This is a use-after-free waiting to happen -- `StringRef` is a non-owning object.

Reply:
Should KernelArgName be a std::unique_ptr?